File: expedia-server/booking-server/src/test/java/org/clonecoder/bookingserver/acceptance/AcceptanceTest.java
package org.clonecoder.bookingserver.acceptance;

import io.restassured.RestAssured;
import org.clonecoder.bookingserver.BookingTest;
import org.junit.jupiter.api.BeforeEach;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.web.server.LocalServerPort;

public class AcceptanceTest extends BookingTest {
    @LocalServerPort
    private int port;

    @BeforeEach
    public void setUp() {
        RestAssured.port = port;
    }
}

File: authentication-service/src/test/kotlin/com/expedia/authentication/acceptance/AcceptanceTest.kt
package com.expedia.authentication.acceptance

import com.expedia.authentication.support.DatabaseCleanup
import io.restassured.RestAssured
import org.junit.jupiter.api.BeforeEach
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.web.server.LocalServerPort

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class AcceptanceTest {

    @Autowired lateinit var databaseCleanup: DatabaseCleanup

    @LocalServerPort
    lateinit var port: String

    @BeforeEach
    fun setUp() {
        RestAssured.port = port.toInt()
        databaseCleanup.execute()
    }
}